sway-core/src/submodules/dependency_graph.rs

```diff
@@ -0,0 +1,87 @@
+use std::collections::HashMap;
+
+use petgraph::{stable_graph::StableGraph, Directed};
+use sway_types::Span;
+
+use crate::language::{ty::TyProgram, DepName};
+
+type GraphIx = u32;
+type NodeIx = petgraph::graph::NodeIndex<GraphIx>;
+
+#[derive(Debug, Hash, PartialEq, Eq, Clone)]
+pub(crate) struct Node {
+    pub(crate) dep_decls: Vec<NodeIx>,
```

Comment:
If we're to go the graph route, we probably want to represent the `dep_decls` on the edges, rather than storing a separate list on the node.